routes/categories.js

```diff
@@ -1,75 +1,109 @@
-    .catch((err)=>{res.status(400).send("Error occured while fetching catagory details")});
-})
+                return res.json({ msg: "Category fetched successfully", data: category })
+            })
+            .catch(err => {
+                console.error(`⚡[server][categoryRoute][get /:id] Error :: ${err}`);
+                return res.status(500).json({ msg: "Internal server error" });
+            });
+    }
+);
 
-router.route("/add").post((req,res)=>
-{
-    console.log(req.body);
-    
-    var newCategory = new Category(
-        {
-            categoryName:req.body.category,
-            categoryImage:req.body.categoryImage,
-            subCategoryList:req.body.subCategoryList
-        }
-    );
-
-    newCategory.save()
-    .then(() => res.json("category added"))
-    .catch((err) => res.status(400).json("Error"+err));
-});
+// route    :: POST /api/categories
+// desc     :: Create a new category
+// access   :: Admin
+router.post(
+    '/',
+    passport.authenticate('jwt', { session: false }),
+    checkIsAdmin,
+    validateCategory,
+    (req, res) => {
+        const newCategory = new Category({
+            categoryName: req.body.categoryName,
+            categoryImage: req.body.categoryImage,
+            subCategoryList: req.body.subCategoryList
+        });
+
+        newCategory.save()
+            .then(category => res.json({ msg: "Category created successfully", data: category }))
+            .catch(err => {
+                console.error(`⚡[server][categoryRoute][post /] Error :: ${err}`);
+                return res.status(500).json({ msg: "Internal server error" });
+            });
+    }
+);
+
+
+router.route("/getCategoryDetails/:categoryName").get((req, res) => {
+    Category.findOne({ categoryName: req.params.categoryName })
+        .then((categories) => { res.send(categories) })
+        .catch((err) => { res.status(400).send("Error occured while fetching catagory details") });
+})
 
-router.route("/getCategories").get((req,res) =>
-{
-    Category.find({},{_id:0,categoryName:1})
-    .then((categories) => res.json(categories))
-    .catch((err) => {res.status(400).json("Error while fetching categories:"+err)});
+router.route("/getCategories").get((req, res) => {
+    Category.find({}, { _id: 0, categoryName: 1 })
+        .then((categories) => res.json(categories))
+        .catch((err) => { res.status(400).json("Error while fetching categories:" + err) });
 });
 
-router.route("/updateCategoryDetails/").put((req,res) =>
-{
-    Category.updateOne({categoryName:req.body.category},{$set:{subCategoryList:req.body.subCategoryList,categoryImage:req.body.categoryImage}})
-    .then(()=>{res.json("Category updated successfully")})
-    .catch(()=>{res.status(400).json("Error:"+err)})
+router.route("/updateCategoryDetails/").put((req, res) => {
+    Category.updateOne({ categoryName: req.body.category }, { $set: { subCategoryList: req.body.subCategoryList, categoryImage: req.body.categoryImage } })
```

Comment:
is there a change where either `req.body.subCategoryList` or `req.body.subCategoryList` be undefined ? and if so, can it lead to unpredicatable behaviour for this api ?

Reply:
router.route("/updateCategoryDetails/").put((req, res) => {}

this route will be deletd

Reply:
got it